docs/src/charmil_config.md

```diff
@@ -0,0 +1,180 @@
+      **Important**: Every field of the defined struct needs to be exportable (ie. start with an uppercase letter). The unexportable fields cannot be stored as config.
+
+            _Example:_
+
+            ```go
+          	type config struct {
+          		Key1 string
+          		Key2 string
+          		Key3 string
+          		Key4 string
+          	}
+
+          	cfg = &config{}
+            ```
+
+  4.  Store a new instance of the Charmil Config handler by calling the `NewHandler` function while passing the path of local config file and the instance of the config struct (initialized in the last step) as arguments.
+
+      _Example:_
+
+      ```go
+      h = c.NewHandler("./examples/host/config.json", cfg)
+      ```
+
+  5.  Load config values from the local config file using the `Load` method.
+
+      _Example:_
+
+      ```go
+      err := h.Load()
+      if err != nil {
+      	log.Fatal(err)
+      }
+      ```
+
+  6.  You can set/get/modify values under any key of config using the idiomatic way to interact with structs in Golang.
+
+      _Example:_
+
+      ```go
+      // Sets a value into config
+      cfg.Key4 = "val4"
+
+      // Overwrites a value in config
+      cfg.Key2 = "newVal2"
+
+      // Returns the value under specified key in config
+      fmt.Println(cfg.Key3) // Prints: val3
+      ```
+
+  7.  Write current config into the local config file using the `Save` method.
+
+          _Example:_
+
+          ```go
+          err = h.Save()
+          if err != nil {
+          	log.Fatal(err)
+          }
+          ```
+
+- ### For plugin developers:
+
+  1.  Open the file where the root command of your CLI is defined using Cobra.
+
+  2.  Import the Charmil Config package by adding the following line at the top of that file:
+      ```go
+      import c "github.com/aerogear/charmil/core/config"
+      ```
+  3.  Define a struct whose fields represent the keys to all the values that you want to store as config and create an instance of it.
+      **Important**: Every field of the defined struct needs to be exportable (ie. start with an uppercase letter). The unexportable fields cannot be stored as config.
+
+            _Example:_
+
+            ```go
+          	type config struct {
+          		Key5 string
+          		Key6 string
+          		Key7 string
+          		Key8 string
+          	}
+
+          	cfg = &config{}
+            ```
+
+  4.  You can set/get/modify values under any key of config using the idiomatic way to interact with structs in Golang.
+
+      _Example:_
+
+      ```go
+      // Sets values into config
+      cfg.Key5 = "val5"
+      cfg.Key6 = "val6"
+      cfg.Key7 = "oldVal7"
+      cfg.Key8 = "val8"
+
+      // Overwrites a value in config
+      cfg.Key7 = "val7"
+
+      // Returns the value under specified key in config
+      fmt.Println(cfg.Key6) // Prints: val6
+      ```
+
+  5.  Use the `MergePluginCfg` function to merge the current plugin config into the local config file
```

Comment:
So far all was really good, but I feel that this merge can be done by including structure of the plugin config inside host config. This way we get serialization sorted out of the box. (Requirement: it needs to be json is fine for now)

Reply:
Resolved via commit: https://github.com/aerogear/charmil/pull/108/commits/a33727536fea8cbaafeae25b96ce55eaab12ef72